ThreeBotPackages/tfgrid_solutions/tfgrid_solutions/chatflows/monitoring_solution_deploy.py

```diff
@@ -0,0 +1,340 @@
+from Jumpscale import j
+import math
+
+
+class MonitoringSolutionDeploy(j.servers.chatflow.get_class()):
+    steps = [
+        "deployment_start",
+        "network_selection",
+        "solution_name",
+        "public_key_get",
+        "prometheus_container_resources",
+        "prometheus_volume_details",
+        "grafana_container_resources",
+        "redis_container_resources",
+        "container_node_id",
+        "farm_selection",
+        "prometheus_container_ip",
+        "grafana_container_ip",
+        "redis_container_ip",
+        "expiration_time",
+        "overview",
+        "containers_pay",
+        "success",
+    ]
+
+    @j.baseclasses.chatflow_step()
+    def deployment_start(self):
+        self.user_info = self.user_info()
+        j.sal.reservation_chatflow.validate_user(self.user_info)
+        self.user_form_data = {}
+        self.user_form_data["chatflow"] = "monitoring"
+        self.md_show(
+            "# This wizard will help you deploy a monitoring system that includes Prometheus, Grafana, and redis",
+            md=True,
+        )
+        self.env_var_dict = {}
+        self.prometheus_query = {}
+        self.grafana_query = {}
+        self.redis_query = {}
+        self.query = {"Prometheus": self.prometheus_query, "Grafana": self.grafana_query, "Redis": self.redis_query}
+        self.ip_addresses = {"Prometheus": "", "Grafana": "", "Redis": ""}
+
+    @j.baseclasses.chatflow_step(title="Network")
+    def network_selection(self):
+        self.network = j.sal.reservation_chatflow.network_select(self, j.me.tid)
+
+    @j.baseclasses.chatflow_step(title="Solution name")
+    def solution_name(self):
+        self.model = j.threebot.packages.tfgrid_solutions.tfgrid_solutions.bcdb_model_get(
+            "tfgrid.solutions.monitoring.1"
+        )
+        self.user_form_data["Solution name"] = j.sal.reservation_chatflow.solution_name_add(self, self.model)
+
+    @j.baseclasses.chatflow_step(title="Access keys")
+    def public_key_get(self):
+        self.user_form_data["Public key"] = self.upload_file(
+            """Please add your public ssh key, this will allow you to access the deployed containers using ssh.
+                    Just upload the file with the key""",
+            required=True,
+        ).split("\n")[0]
+
+    @j.baseclasses.chatflow_step(title="Prometheus container resources")
+    def prometheus_container_resources(self):
```

Comment:
Are the proposed default value for CPU and memory viable for a Prometheus container ? 1GB seems very very few for a db like that...

Reply:
is 4GB reasonable enough as a default?

Reply:
well I don't know exactly, But I guess we can find recommendation on their website.

Reply:
According to this https://prometheus.io/docs/prometheus/1.8/storage/#memory-usage 3Gib would work as a default

Reply:
updated here https://github.com/threefoldtech/jumpscaleX_threebot/pull/764/commits/2c57640372d6d533176a56676396ab87810b0f3b#diff-0e24746aa7feb62e0d805614631f5de0R68